syncstorage/storage/sql/__init__.py

```diff
@@ -460,6 +462,72 @@ def set_items(self, session, userid, collection, items):
         return self._touch_collection(session, userid, collectionid)
 
     @with_session
+    def create_batch(self, session, userid, collection):
+        """Creates a batch in batch_uploads table"""
+        collectionid = self._get_collection_id(session, collection)
+        batchid = ts2bigint(session.timestamp)
+        params = {
+            "batch": batchid,
+            "userid": userid,
+            "collection": collectionid
+        }
+        session.query("CREATE_BATCH", params)
+        return batchid
+
+    @with_session
+    def valid_batch(self, session, userid, collection, batchid):
+        """Checks to see if the batch ID is valid and still open"""
+        collectionid = self._get_collection_id(session, collection)
+        params = {
+            "batch": batchid,
+            "userid": userid,
+            "collection": collectionid
+        }
+        valid = session.query_scalar("VALID_BATCH", params=params)
+        return valid
+
+    @with_session
+    def append_items_to_batch(self, session, userid, collection, batchid,
+                              items):
+        """Inserts items into batch_upload_items"""
+        rows = []
+        for data in items:
+            id_ = data["id"]
+            row = self._prepare_bui_row(session, batchid, id_, data)
+            rows.append(row)
+        defaults = {
+            "modified": ts2bigint(session.timestamp),
+            "payload": "",
+            "payload_size": 0,
+        }
+        session.insert_or_update("batch_upload_items", rows, defaults)
```

Comment:
I wish there were an obvious way to check that the size of the batch has not gotten too big...but I can't think of anything better than a "SELECT COUNT()" and I don't think we want to pay the overhead of that.